mobile/examples/cc/fetch_client/fetch_client.cc

```diff
@@ -55,45 +58,39 @@ void Fetch::sendRequest(const absl::string_view url_string) {
     absl::MutexLock lock(&engine_mutex_);
     stream_prototype = engine_->streamClient()->newStreamPrototype();
   }
-  stream_prototype->setOnHeaders(
-      [](Platform::ResponseHeadersSharedPtr headers, bool, envoy_stream_intel intel) {
-        std::cerr << "Received headers on connection: " << intel.connection_id << "\n";
-
-        const Platform::RawHeaderMap raw_headers = headers->allHeaders();
-        for (const auto& [key, header] : raw_headers) {
-          for (const auto& val : header) {
-            std::cout << key << ": " << val << "\n";
-          }
-        }
-      });
-  stream_prototype->setOnData([](envoy_data c_data, bool fin) {
-    std::cout << Data::Utility::copyToString(c_data);
-    if (fin) {
-      std::cout << "Received final data\n";
+  EnvoyStreamCallbacks stream_callbacks;
+  stream_callbacks.on_headers_ = [&](const Http::ResponseHeaderMap& headers, bool /* end_stream */,
```

Comment:
why do we need to capture all the variables by ref here? (i.e. `[&]`)  same question for below

Reply:
Removed.